### ISSUE TITLE
Create circuite breaker for API

### ISSUE THREAD

**Issue by acbeni** [commercetools/commercetools-jvm-sdk#1630]
Sometimes when badely impemented some clientd try to repeatedly reconnect to our backend, we can protect ourselves from this by adding a circuit breaker to our API

**Comment by acbeni** [comment]
resilience4j seems like a promising choice https://github.com/resilience4j/resilience4j

**Comment by katmatt** [comment]
I once used hystrix as a circuit breaker [https://github.com/Netflix/Hystrix](https://github.com/Netflix/Hystrix) and really liked it.

**Comment by acbeni** [comment]
Yep it's the most famous one but resilience4j i think is more lightweight. still its good that you mentioned  it @katmatt to make a small comparative study before choosing one 👍

**Comment by acbeni** [comment]
After examining the choice, it seems a bit more constraining and also it means more dependencies for our API, and if a client wants it, it's a matter of creating a new decorator.
i would rather let it as simple as it is right now. 
WDYT @katmatt ?
